source/adapters/cuda/event.cpp

```diff
@@ -220,13 +220,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventSetCallback(ur_event_handle_t,
 UR_APIEXPORT ur_result_t UR_APICALL
 urEventWait(uint32_t numEvents, const ur_event_handle_t *phEventWaitList) {
   try {
-    auto Context = phEventWaitList[0]->getContext();
-    ScopedContext Active(Context);
-
-    auto WaitFunc = [Context](ur_event_handle_t Event) -> ur_result_t {
+    auto WaitFunc = [](ur_event_handle_t Event) -> ur_result_t {
       UR_ASSERT(Event, UR_RESULT_ERROR_INVALID_EVENT);
-      UR_ASSERT(Event->getContext() == Context,
-                UR_RESULT_ERROR_INVALID_CONTEXT);
+      ScopedContext Active(Event->getContext());
```

Comment:
I don't think we need to set the context every time this lambda is run. It's valid to wait on an event that was recorded in another context, so setting the context just once at the start of the `try catch` should be enough

Reply:
Is it valid to wait for an event that was set in another CUcontext (on a different device)?
I assumed this was not the case.